File: fix_suspense.py
import os

tools_dir = 'src/app/[locale]'

for root, dirs, files in os.walk(tools_dir):
    for file in files:
        if file == 'page.tsx':
            page_path = os.path.join(root, file)
            with open(page_path, 'r') as f:
                content = f.read()
            
            if 'import Client from "./Client";' in content or "import Client from './Client';" in content:
                if '<Client />' in content and '<Suspense' not in content:
                    if 'import { Suspense }' not in content:
                        content = 'import { Suspense } from "react";\n' + content
                    
                    content = content.replace('<Client />', '<Suspense fallback={<div className="tool-grid card" style={{textAlign: "center", padding: "40px"}}>Loading calculator...</div>}>\n\t\t\t\t<Client />\n\t\t\t</Suspense>')
                    
                    with open(page_path, 'w') as f:
                        f.write(content)
                    print(f"Fixed {page_path}")

File: scripts/fix-suspense.mjs
import fs from 'fs';
import path from 'path';

const localeDir = path.resolve('src/app/[locale]');
const dirs = fs.readdirSync(localeDir, { withFileTypes: true })
    .filter(dirent => dirent.isDirectory())
    .map(dirent => dirent.name);

for (const dir of dirs) {
    const pagePath = path.join(localeDir, dir, 'page.tsx');
    if (!fs.existsSync(pagePath)) continue;

    let content = fs.readFileSync(pagePath, 'utf8');
    
    if (content.includes('<Suspense')) {
        continue;
    }

    if (!content.includes('import { Suspense }')) {
        content = 'import { Suspense } from "react";\n' + content;
    }

    const returnRegex = /return\s+<([a-zA-Z0-9_]+Client)\s*\/>;/g;
    content = content.replace(returnRegex, 'return <Suspense fallback={<div className="loading">Loading...</div>}><$1 /></Suspense>;');
    
    const returnPropsRegex = /return\s+<([a-zA-Z0-9_]+Client)([^>]*)>;/g;
    content = content.replace(returnPropsRegex, 'return <Suspense fallback={<div className="loading">Loading...</div>}><$1$2></Suspense>;');

    fs.writeFileSync(pagePath, content, 'utf8');
}
console.log('Fixed Suspense in page.tsx files');
